mars_troughs/mcmc.py

```diff
@@ -0,0 +1,239 @@
+            self.lag_model_name=auxLag.split('.')
+            self.lag_model_name=self.lag_model_name[2]
+        
+        #Create directory to save outputs
+        if not os.path.exists(self.directory):
+            os.makedirs(self.directory)
+        #Create directory to save objects and h5 tracking file
+        if not os.path.exists(self.directory+'obj/'):
+            os.makedirs(self.directory+'obj/')
+            
+        self.modelName=self.acc_model_name+'_'+self.lag_model_name
+        #if not os.path.exists(self.directory+'obj/'+self.modelName+'/'):
+         #   os.makedirs(self.directory+'obj/'+self.modelName+'/')
+    
+        #self.filename=self.directory+'obj/'+self.modelName+'/'+str(self.maxSteps)
+        self.filename=self.directory+'obj/'+self.modelName+'_'+str(self.maxSteps)+'obj'
+        
+        #Set optimized parameter values as initial values of MCMC chains 
+        self.initParams=np.array([self.optParams+ 
+                        1e-3*self.optParams*np.random.randn(self.ndim) 
+                        for i in range(self.nwalkers)])
+        
+        #check that all initParams fit the priors
+        likeInit=np.zeros((self.nwalkers,1))
+        for i in range(0,self.nwalkers):
+            iparams=dict(zip(self.tr.all_parameter_names,self.initParams[i,:]))
+            likeInit[i]=self.ln_likelihood(iparams)
+        
+        indxProblem=[i for i, n in enumerate(likeInit) if n == -1e99]
+        
+        if len(indxProblem)>0:
+            for i in range(0,len(indxProblem)):
+                like=-1e99
+                while like==-1e99:
+                    iparamsTest=(self.optParams+ 1e-3*self.optParams*
+                                 np.random.randn(self.ndim))
+                    iparamsDict=dict(zip(self.tr.all_parameter_names,iparamsTest))
+                    like=self.ln_likelihood(iparamsDict)
+                self.initParams[indxProblem[i],:]=iparamsTest
+                
+        sys.stderr.flush()  
+        print('All initial parameters fit priors',file=sys.stderr)
+        sys.stderr.flush()
+
+        start = time.time()
+        
+        #Initialize sampler
+        self.sampler = emcee.EnsembleSampler(self.nwalkers, self.ndim, 
+                                             self.ln_likelihood,  
+                                             parameter_names=self.parameter_names)
+        #Run MCMC and track progress
+        self.sampler.reset()
+        #Iteratively compute autocorrelation time Tau
+        index=0
+        self.autocorr=np.zeros(10)
+        self.taus=np.zeros((10,self.ndim))
+        old_tau=np.inf
+        
+        #compute tau every n iterations
+        for sample in self.sampler.sample(self.initParams,iterations=self.maxSteps, 
+        progress=False):
+        
+            if self.sampler.iteration%(self.maxSteps/10):
+                continue
+            sys.stderr.flush()
+            print(self.sampler.iteration/self.maxSteps*100,'%',file=sys.stderr)
```

Comment:
These printing calls may be a large source of slowdown for the MCMC scripts. I'd recommend not using `print` to save results, but instead using other methods like a regular rile writer
```python
with open("myfile.txt", "w") as fh:
     fh.write("stuff")
```